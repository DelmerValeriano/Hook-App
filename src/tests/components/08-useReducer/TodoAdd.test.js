import React from 'react';
import {shallow} from 'enzyme';
import { TodoAdd } from '../../../components/08-useReducer/TodoAdd';


describe('pruebas en el <TodoAdd/>', () => {  

    const handleAddTodo =jest.fn();

    const wrapper = shallow(<TodoAdd
        handleAddTodo={handleAddTodo}
    />)

    test('debe de mostrarse correctamente', () => {  
        expect(wrapper).toMatchSnapshot();

    });

    test('No debe de llamar al handleAddTodo', () => {  
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({preventDefault(){}});
        expect(handleAddTodo).toHaveBeenCalledTimes(0);
       
    });
    test('debe de llamar al handleAddTodo', () => {  
        const value ='Aprende react';
        wrapper.find('input').simulate('change',{
            target:{
                value,
                name: 'description'
            }
        });
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({preventDefault(){}});
        expect(handleAddTodo).toHaveBeenCalledTimes(1);
        expect(handleAddTodo).toBeCalledWith(expect.any(Object));
        expect(handleAddTodo).toBeCalledWith({
            id: expect.any(Number),
            desc:value,
            done:false,

        });
        expect(wrapper.find('input').prop('value')).toBe('');





       
    });


})
