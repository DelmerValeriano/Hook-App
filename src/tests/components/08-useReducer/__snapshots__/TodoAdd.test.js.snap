// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <TodoAdd/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <div
      className="d-grid"
    >
      <button
        className="btn btn-outline-primary mt-2 btn-block"
        type="submit"
      >
        Agregar
      </button>
    </div>
  </form>
</Fragment>
`;
